### PR TITLE
[DNM] Trying out code sharing approach for application modification

### PR DESCRIPTION
**Pull request by JeroenDeDauw**
Based on https://github.com/wmde/FundraisingStore/pull/90

---

Goal: avoid boilerplate code in services that need to modify a membership application identified by ID, and simplify testing them.

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
@KaiNissen I added both the callable approach and the individual methods to the class in the store component. Opted to go with the callable here, as then testing the failure cases is real easy: just one test in which the modifyApplication throws an exception. When using the individual methods, you need to test an exception for each, and need to verify the second one actually gets called.

File: src/DataAccess/DoctrineMembershipApplicationPiwikTracker.php [hunk 1]
```diff
@@ -4,55 +4,35 @@
...
-	private function persistApplication( MembershipApplication $application ) {
 		try {
-			$this->entityManager->persist( $application );
-			$this->entityManager->flush();
+			$this->applicationModifier->modifyApplication(
+				$applicationId,
+				function( MembershipApplication $application ) use ( $trackingString ) {
+					$application->setTracking( $trackingString );
+				}
+			);
```
Comment (JeroenDeDauw) [inline src/DataAccess/DoctrineMembershipApplicationPiwikTracker.php]: Should also call `$this->applicationModifier->flush()` the way it is currently implemented

**Comment by JeroenDeDauw** [comment]
Any thoughts on this?

**Comment by JeroenDeDauw** [comment]
Ping @gbirke @KaiNissen @wiese <o>

**Comment by timEulitz** [comment]
This PR is almost 2 years old, is there any action that could be taken in the near future / should we pursue this further?

**Comment by gbirke** [comment]
Was implemented by https://github.com/wmde/fundraising-memberships/pull/5

**Pull request closed without merging**